src/keybaseca/bot/bot.go

```diff
@@ -18,42 +22,47 @@ import (
 	log "github.com/sirupsen/logrus"
 )
 
-// Get a running instance of the keybase chat API. Will use the configured credentials if necessary.
-func GetKBChat(conf config.Config) (*kbchat.API, error) {
-	return botwrapper.GetKBChat(conf.GetKeybaseHomeDir(), conf.GetKeybasePaperKey(), conf.GetKeybaseUsername(), conf.GetKeybaseTimeout())
+// Bot is a SSH CA Keybase-backed bot
+type Bot struct {
+	conf config.Config
+	api  *kbchat.API
 }
 
-// Get the username of the user that the keybaseca bot is running as
-func GetUsername(conf config.Config) (string, error) {
-	kbChat, err := GetKBChat(conf)
+// NewBot creates a new Keybase chat API with the given config
+func NewBot(conf config.Config) (bot Bot, err error) {
+	api, err := botwrapper.GetKBChat(conf.GetKeybaseHomeDir(), conf.GetKeybasePaperKey(), conf.GetKeybaseUsername(), conf.GetKeybaseTimeout())
 	if err != nil {
-		return "", fmt.Errorf("failed to start Keybase chat: %v", err)
+		return bot, fmt.Errorf("error starting Keybase chat: %v", err)
 	}
-	username := kbChat.GetUsername()
-	if username == "" {
-		return "", fmt.Errorf("failed to get a username from kbChat, got an empty string")
-	}
-	return username, nil
+	return Bot{conf: conf, api: api}, nil
 }
 
-// Start the keybaseca bot in an infinite loop. Does not return unless it encounters an unrecoverable error.
-func StartBot(conf config.Config) error {
-	kbc, err := GetKBChat(conf)
+// Start the SSH CA bot in an infinite loop. Does not return unless it
+// encounters an unrecoverable error.
+func (b *Bot) Start() error {
+	err := b.writeClientConfig()
```

Comment:
i moved the writing and deleting (and capturing SIGTERM) of the client config from src/cmd/keybaseca/keybaseca.go to here (in bot.go)

Reply:
it's also not really relevant to the caller - which is why i assumed you pushed it down here. but both reasons seem good.